artman/cli/main2.py

```diff
@@ -0,0 +1,435 @@
+        logger.warn('This is probably your first time running Artman.')
+        logger.warn('Run `configure-artman` to get yourself set up.')
+        sys.exit(64)
+
+    # Done; return the user config.
+    return user_config
+
+
+def normalize_flags(flags, user_config):
+    """Combine the argparse flags and user configuration together.
+
+    Args:
+        flags (argparse.Namespace): The flags parsed from sys.argv
+        user_config (dict): The user configuration taken from
+                            ~/.artman/config.yaml.
+
+    Returns:
+        tuple (str, dict): 2-tuple containing:
+            - pipeline name
+            - pipeline arguments
+    """
+    flags.input_dir = os.path.abspath(flags.input_dir)
+    flags.output_dir = os.path.abspath(flags.output_dir)
+    flags.config = os.path.abspath(flags.config)
+    pipeline_args = {}
+
+    # Determine logging verbosity and then set up logging.
+    verbosity = support.resolve('verbosity', user_config, flags, default=INFO)
+    setup_logging(verbosity)
+
+    # Save local paths, if applicable.
+    # This allows the user to override the path to api-client-staging or
+    # toolkit on his or her machine.
+    pipeline_args['local_paths'] = support.parse_local_paths(
+        user_config, flags.input_dir)
+
+    artman_config_path = flags.config
+    if not os.path.isfile(artman_config_path):
+        logger.error(
+            'Artman config file `%s` doesn\'t exist.' % artman_config_path)
+        sys.exit(96)
+
+    try:
+        artifact_config = loader.load_artifact_config(
+            artman_config_path, flags.artifact_name, flags.input_dir)
+    except ValueError as ve:
+        logger.error('Artifact config loading failed with `%s`' % ve)
+
+    # If we were given just an API or BATCH, then expand it into the --config
+    # syntax.
+    shared_config_name = 'common.yaml'
+    if artifact_config.language == config_pb2.Artifact.RUBY:
+        # TODO(ethanbao): Figure out why.
+        shared_config_name = 'doc.yaml'
+
+    legacy_config_dict = converter.convert_to_legacy_config_dict(
+        artifact_config, flags.input_dir, flags.output_dir)
+    logger.debug('Below is the legacy config after conversion:\n%s' %
+                 pprint.pformat(legacy_config_dict))
+    tmp_legacy_config_yaml = '%s.tmp' % artman_config_path
+    with open(tmp_legacy_config_yaml, 'w') as outfile:
+        yaml.dump(legacy_config_dict, outfile, default_flow_style=False)
+
+    googleapis = os.path.realpath(
+        os.path.expanduser(
+            pipeline_args['local_paths']['googleapis'], ))
+    config = ','.join([
+        '{artman_config_path}',
+        '{googleapis}/gapic/lang/{shared_config_name}',
+    ]).format(
+        artman_config_path=tmp_legacy_config_yaml,
+        googleapis=googleapis,
+        shared_config_name=shared_config_name, )
+
+    # Set the pipeline as well as package_type and packaging
+    artifact_type = artifact_config.type
```

Comment:
This is just a minor readability item; it would be easier to understand (for me) if each final destination had one predicate, instead of having three destinations and five predicates.